src/network/networkd-link.c

```diff
@@ -1749,10 +1749,12 @@ void link_update_operstate(Link *link, bool also_update_master) {
          *                 ------------------------------------------------------------------------------
          *                 off      | off | no-carrier | dormant | degraded-carrier | carrier  | enslaved
          * address_state   degraded | off | no-carrier | dormant | degraded-carrier | degraded | enslaved
```

Comment:
I think we should also change this line degraded-carrier -> degraded.

Reply:
you wrote "this line" but I guess that you mean both lines? aka so that the state is degraded if address_state is off or degraded when the carrier_state is degraded-carrier so the only case where the operational state is degraded-carrier is when carrier_state is degraded-carrier and address_state is routable? Or do you want to change just the address_state=degraded && carrier_state=degraded-carrier to be operational_state==degraded?

Reply:
done that now